
# Project: The Olympics Games

## Table of Contents
<ul>
<li><a href="#intro">Introduction</a></li>
<li><a href="#Checking Athletes Dataset">Checking Athletes Dataset</a></li>
<li><a href="#Checking Teams Dataset">Checking Teams Datase</a></li>
<li><a href="#Checking Medals Dataset">Checking Medals Dataset</a></li>
<li><a href="#Checking Coaches Dataset">Checking Coaches Dataset</a></li>
<li><a href="#Checking Gender Dataset">Checking Gender Dataset</a></li>
<li><a href="#Checking Country Medals from 1896 to 2020 Dataset">Checking Country_Medals from_1896 to 2020 Dataset</a></li>
<li><a href="#conclusions">Conclusions</a></li>
</ul>

<a id='intro'></a>
## Introduction

> I will analyze and explore data from 6 Datasets, 5 Datasets (Athletes, Teams, Medals, Coaches, and Gender) for Tokyo 2020. And one dataset (Country_Medals_from_1896_to_2020) for the history of the olympics games from 1896 to 2020.
In this notebook, I will check the data cleaning and consistency then I will the exploratory and Explanatory Processes through Tableau and I will include the insights I found In the notebook.

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sb
%matplotlib inline
from scipy import stats
import datetime

<a id='Checking Athletes Dataset'></a>
## Checking Athletes Dataset

In [2]:
athletes = pd.read_csv('Athletes.csv')
print(athletes.shape)
athletes.sample(5)

(11085, 3)


Name       Country                   Play
2457              DOUMBIA Idrissa  Cote d'Ivoir               Football
1994               CREVIER Axelle        Canada             Water Polo
8033  RAMIREZ LIMON Andrea Soraya        Mexico              Athletics
3610                 GUSTIN Ronan        France                Fencing
9129               SMITH Samantha        Canada  Trampoline Gymnastics

In [3]:
athletes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11085 entries, 0 to 11084
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Name     11085 non-null  object
 1   Country  11085 non-null  object
 2   Play     11085 non-null  object
dtypes: object(3)
memory usage: 259.9+ KB


In [4]:
athletes.isnull().sum()

Name       0
Country    0
Play       0
dtype: int64

In [5]:
athletes.duplicated().sum()

1

In [6]:
athletes[athletes.duplicated()]

Name  Country      Play
254  ALI Mohamed  Bahrain  Handball

- Duplicate athlete.

In [7]:
athletes.Name.value_counts(ascending = False).head(20)

MARTIN Daniel           2
WANG Yang               2
KIM Hyunsoo             2
DYGERT Chloe            2
PALTRINIERI Gregorio    2
GANNA Filippo           2
HALL James              2
KURBANOV Ruslan         2
SUN Jiajun              2
ALVAREZ Jorge           2
PEREZ Paola             2
PEREZ Maria             2
HAVIK Yoeri             2
WELLBROCK Florian       2
KOPECKY Lotte           2
WATANABE Yuta           2
ALI Mohamed             2
ZHANG Xin               2
LI Qian                 2
CHEN Yang               2
Name: Name, dtype: int64

In [8]:
athletes[athletes['Name'] == 'MARTIN Daniel']

Name  Country          Play
6044  MARTIN Daniel  Ireland  Cycling Road
6045  MARTIN Daniel  Romania      Swimming

In [9]:
athletes[athletes['Name'] == 'ALVAREZ Jorge']

Name   Country      Play
319  ALVAREZ Jorge      Cuba  Shooting
320  ALVAREZ Jorge  Honduras  Football

In [10]:
athletes[athletes['Name'] == 'van ROUWENDAAL Sharon']

Name      Country               Play
10111  van ROUWENDAAL Sharon  Netherlands  Marathon Swimming
10112  van ROUWENDAAL Sharon  Netherlands           Swimming

In [11]:
athletes[athletes['Name'] == 'KOPECKY Lotte']

Name  Country           Play
5061  KOPECKY Lotte  Belgium   Cycling Road
5062  KOPECKY Lotte  Belgium  Cycling Track

In [12]:
athletes[athletes['Name'] == 'HALL James']

Name                   Country                 Play
3664  HALL James             Great Britain  Artistic Gymnastics
3665  HALL James  United States of America             Shooting

In [13]:
athletes[athletes['Name'] == 'LI Qian']

Name                     Country          Play
5522  LI Qian  People's Republic of China        Boxing
5523  LI Qian                      Poland  Table Tennis

In [14]:
athletes[athletes['Name'] == 'ZHANG Xin']

Name                     Country           Play
11016  ZHANG Xin  People's Republic of China       Football
11017  ZHANG Xin  People's Republic of China  Skateboarding

In [15]:
athletes.Country.value_counts(ascending = True).head(10)

St Vincent and the Grenadines    2
Bermuda                          2
Marshall Islands                 2
Saint Kitts and Nevis            2
United Republic of Tanzania      2
Mauritania                       2
Dominica                         2
Vanuatu                          2
Myanmar                          2
Lesotho                          2
Name: Country, dtype: int64

In [80]:
athletes.Play.value_counts(ascending = True).head(10)

Cycling BMX Freestyle    19
Trampoline Gymnastics    31
Sport Climbing           37
Surfing                  38
Cycling BMX Racing       43
Marathon Swimming        49
3x3 Basketball           62
Modern Pentathlon        69
Cycling Mountain Bike    74
Skateboarding            77
Name: Play, dtype: int64

In [17]:
# Removing the duplicated record

athletes = athletes.drop_duplicates()

In [18]:
athletes.shape

(11084, 3)

In [78]:
athletes.Name.nunique()

11062

In [77]:
athletes.Play.nunique()

46

In [19]:
# Storing the cleaed data after removing the duplicates

athletes.to_csv('Cleaned_athletes.csv', index=False)

<a id='Checking Teams Dataset'></a>
## Checking Teams Dataset

In [20]:
teams = pd.read_csv('Teams.csv')
print(teams.shape)
teams.sample(5)

(743, 3)


Name                 Play               Event
38        India              Archery          Mixed Team
481  Azerbaijan  Rhythmic Gymnastics    Group All-Around
88        Spain  Artistic Gymnastics        Women's Team
396     Tunisia              Fencing  Women's Sabre Team
392         ROC              Fencing        Women's ��p�

In [21]:
teams.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 743 entries, 0 to 742
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Name    743 non-null    object
 1   Play    743 non-null    object
 2   Event   743 non-null    object
dtypes: object(3)
memory usage: 17.5+ KB


In [71]:
teams.Name.nunique()

146

In [74]:
teams.Play.nunique()

20

In [22]:
teams.isnull().sum()

Name     0
Play     0
Event    0
dtype: int64

In [23]:
teams.duplicated().sum()

0

In [73]:
teams.Name.value_counts(ascending = True)

Ishii/Murakami           1
Huberli/Betschart        1
WANG Zhen / ZHANG Mo     1
Venezuela                1
Plavins/Tocs             1
                        ..
France                  32
Australia               32
Italy                   34
United States           43
Japan                   45
Name: Name, Length: 146, dtype: int64

In [75]:
teams.Play.value_counts(ascending = True)

Baseball/Softball       12
Rhythmic Gymnastics     14
3x3 Basketball          16
Triathlon               18
Water Polo              22
Artistic Gymnastics     24
Hockey                  24
Volleyball              24
Rugby Sevens            24
Handball                24
Basketball              24
Football                28
Artistic Swimming       32
Beach Volleyball        48
Table Tennis            48
Fencing                 52
Archery                 53
Cycling Track           64
Athletics               79
Swimming               113
Name: Play, dtype: int64

In [26]:
teams.Event.value_counts(ascending = True).tail(10)

Men's 4 x 200m Freestyle Relay       16
Mixed 4 x 100m Medley Relay          16
Women's 4 x 200m Freestyle Relay     17
Mixed Relay                          18
Duet                                 22
Mixed Team                           29
Women's Team                         40
Men's Team                           40
Women                               114
Men                                 120
Name: Event, dtype: int64

<a id='Checking Medals Dataset'></a>
## Checking Medals Dataset

In [27]:
medals = pd.read_csv('Medals.csv')
print(medals.shape)
medals.sample(5)

(93, 5)


Team  Gold  Silver  Bronze  Total
20      Cuba     7       3       5     15
15     China    38      32      18     88
59   Namibia     0       1       0      1
10  Botswana     0       0       1      1
27  Ethiopia     1       1       2      4

In [28]:
medals.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 93 entries, 0 to 92
Data columns (total 5 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Team    93 non-null     object
 1   Gold    93 non-null     int64 
 2   Silver  93 non-null     int64 
 3   Bronze  93 non-null     int64 
 4   Total   93 non-null     int64 
dtypes: int64(4), object(1)
memory usage: 3.8+ KB


In [29]:
medals.duplicated().sum()

0

In [87]:
medals.describe()

Gold     Silver     Bronze       Total
count  93.000000  93.000000  93.000000   93.000000
mean    3.655914   3.634409   4.322581   11.612903
std     7.022471   6.626339   6.210372   19.091332
min     0.000000   0.000000   0.000000    1.000000
25%     0.000000   0.000000   1.000000    2.000000
50%     1.000000   1.000000   2.000000    4.000000
75%     3.000000   4.000000   5.000000   11.000000
max    39.000000  41.000000  33.000000  113.000000

In [89]:
medals.Total.sum()

1080

In [88]:
medals.Bronze.sum()

402

In [90]:
medals.Gold.sum()

340

<a id='Checking Coaches Dataset'></a>
## Checking Coaches Dataset

In [30]:
coaches = pd.read_csv('Coaches.csv')
print(coaches.shape)
coaches.sample(5)

(394, 3)


Name      Country        Play
17   ARNAU CREUS Xavier        Japan      Hockey
334        SMITH Darren  New Zealand      Hockey
40      BRONDELLO Sandy    Australia  Basketball
241     MAZZANTI Davide        Italy  Volleyball
157        HAVENGA Arno  Netherlands  Water Polo

In [31]:
coaches.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 394 entries, 0 to 393
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Name     394 non-null    object
 1   Country  394 non-null    object
 2   Play     394 non-null    object
dtypes: object(3)
memory usage: 9.4+ KB


In [32]:
coaches.duplicated().sum()

13

In [33]:
coaches[coaches.duplicated()]

Name                     Country               Play
57       CHEPAK Anastasiya                       Egypt  Artistic Swimming
59      CHERNETSKA Natalia                      Greece  Artistic Swimming
93           ELAFANDI Nour                       Egypt  Artistic Swimming
103      FARINELLI Roberta                       Italy  Artistic Swimming
116          FUJIKI Mayuko                       Spain  Artistic Swimming
127  GIALLOMBARDO Patrizia                       Italy  Artistic Swimming
144       GUERRERO Rolando                      Mexico  Baseball/Softball
263       MONTICO Loredana                   Australia  Artistic Swimming
305  REQUENA PERICAS Judit                       Spain  Artistic Swimming
307    RETEGUI Carlos Jose                   Argentina             Hockey
346          SZAUDER Gabor                      Canada  Artistic Swimming
353       TARRES CAMPA Ana                      Greece  Artistic Swimming
376               WANG Jie  People's Republic of China  Artistic Swimming

In [34]:
coaches[coaches['Name'] == 'CHEPAK Anastasiya']

Name Country               Play
56  CHEPAK Anastasiya   Egypt  Artistic Swimming
57  CHEPAK Anastasiya   Egypt  Artistic Swimming

In [35]:
coaches[coaches['Name'] == 'SZAUDER Gabor']

Name Country               Play
345  SZAUDER Gabor  Canada  Artistic Swimming
346  SZAUDER Gabor  Canada  Artistic Swimming

In [36]:
coaches[coaches['Name'] == 'RETEGUI Carlos Jose']

Name    Country    Play
306  RETEGUI Carlos Jose  Argentina  Hockey
307  RETEGUI Carlos Jose  Argentina  Hockey

In [37]:
coaches.Name.value_counts(ascending = True).tail(15)

OLOO Felix               1
SUNDHAGE Pia Mariane     1
RETEGUI Carlos Jose      2
MONTICO Loredana         2
CHEPAK Anastasiya        2
GUERRERO Rolando         2
WANG Jie                 2
CHERNETSKA Natalia       2
ELAFANDI Nour            2
REQUENA PERICAS Judit    2
GIALLOMBARDO Patrizia    2
FUJIKI Mayuko            2
TARRES CAMPA Ana         2
SZAUDER Gabor            2
FARINELLI Roberta        2
Name: Name, dtype: int64

In [38]:
coaches.Country.value_counts(ascending = True).head(20)

Tunisia                     1
Honduras                    1
Cambodia                    1
Bahrain                     1
Turkey                      1
Cote d'Ivoir                1
Liechtenstein               1
Slovakia                    1
Angola                      1
Portugal                    1
Belarus                     1
Croatia                     1
Fiji                        2
Israel                      2
Kazakhstan                  2
Islamic Republic of Iran    2
Romania                     2
Colombia                    2
Denmark                     2
Saudi Arabia                2
Name: Country, dtype: int64

In [39]:
coaches.Play.value_counts(ascending = True).head(10)

Water Polo           22
Rugby Sevens         25
Baseball/Softball    26
Handball             29
Hockey               45
Volleyball           45
Football             59
Artistic Swimming    69
Basketball           74
Name: Play, dtype: int64

In [40]:
# Removing the duplicated record

coaches = coaches.drop_duplicates()
coaches.shape

(381, 3)

In [41]:
# Storing the cleaed data after removing the duplicates

coaches.to_csv('Cleaned_coashes.csv', index=False)

<a id='Checking Gender Dataset'></a>
## Checking Gender Dataset

In [42]:
gender = pd.read_csv('Gender.csv')
print(gender.shape)
gender.sample(5)

(46, 4)


Play  Female  Male  Total
43         Water Polo     122   146    268
36           Swimming     361   418    779
41          Triathlon      55    55    110
8    Beach Volleyball      48    48     96
3   Artistic Swimming     105     0    105

In [43]:
gender.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 46 entries, 0 to 45
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Play    46 non-null     object
 1   Female  46 non-null     int64 
 2   Male    46 non-null     int64 
 3   Total   46 non-null     int64 
dtypes: int64(3), object(1)
memory usage: 1.6+ KB


In [44]:
gender.duplicated().sum()

0

In [45]:
gender.Female.value_counts(ascending = True).tail(10)

10     1
969    1
90     2
192    2
144    2
20     2
86     2
98     2
40     2
96     2
Name: Female, dtype: int64

In [46]:
gender.Total.value_counts(ascending = True).tail(10)

215    1
350    1
130    1
197    1
198    1
288    2
40     2
96     2
82     2
289    2
Name: Total, dtype: int64

In [63]:
gender.Female.sum()

5432

In [64]:
gender.Male.sum()

5884

In [91]:
gender.Total.sum()

11316

In [94]:
gender.Play.nunique()

46

<a id='Checking Country Medals from 1896 to 2020 Dataset'></a>
## Checking Country Medals from 1896 to 2020 Dataset

In [47]:
all_time_country_medals = pd.read_csv('Country_Medals_from_1896_to_2020.csv')
print(all_time_country_medals.shape)
all_time_country_medals.sample(5)

(1344, 1)


Year;Country_Code;Country_Name;Host_city;Host_country;Gold;Silver;Bronze
1092    1992;(TCH);Czechoslovakia;Barcelona;Spain;4;2;1                      
1340            1936;(BEL);Belgium;Berlin;Germany;0;0;2                      
425                1964;(DEN);Denmark;Tokyo;Japan;2;1;3                      
651          2016;;Bulgaria;Rio de Janeiro;Brazil;0;1;2                      
1258       1952;(LUX);Luxembourg;Helsinki;Finland;1;0;0

In [48]:
all_time_country_medals = all_time_country_medals.rename(columns = {'Year;Country_Code;Country_Name;Host_city;Host_country;Gold;Silver;Bronze': 'all_data'}, inplace = False)

In [49]:
'Year;Country_Code;Country_Name;Host_city;Host_country;Gold;Silver;Bronze'.split(';')

['Year',
 'Country_Code',
 'Country_Name',
 'Host_city',
 'Host_country',
 'Gold',
 'Silver',
 'Bronze']

In [50]:
def Year(x):
    return x.split(';')[0]

def Country_Code(x):
    return x.split(';')[1]

def Country_Name(x):
    return x.split(';')[2]

def Host_city(x):
    return x.split(';')[3]

def Host_country(x):
    return x.split(';')[4]

def Gold(x):
    return x.split(';')[5]

def Silver(x):
    return x.split(';')[6]

def Bronze(x):
    return x.split(';')[7]

In [51]:
all_time_country_medals['Year']=all_time_country_medals['all_data'].apply(Year)
all_time_country_medals['Country_Code']=all_time_country_medals['all_data'].apply(Country_Code)
all_time_country_medals['Country_Name']=all_time_country_medals['all_data'].apply(Country_Name)
all_time_country_medals['Host_city']=all_time_country_medals['all_data'].apply(Host_city)
all_time_country_medals['Host_country']=all_time_country_medals['all_data'].apply(Host_country)
all_time_country_medals['Gold']=all_time_country_medals['all_data'].apply(Gold).astype(int)
all_time_country_medals['Silver']=all_time_country_medals['all_data'].apply(Silver).astype(int)
all_time_country_medals['Bronze']=all_time_country_medals['all_data'].apply(Bronze).astype(int)
all_time_country_medals['Total']=all_time_country_medals['Gold']+all_time_country_medals['Silver']+all_time_country_medals['Bronze']

In [52]:
print(all_time_country_medals.shape)
all_time_country_medals.sample(5)

(1344, 10)


all_data  Year Country_Code  \
255    1980;(BRA);Brazil;Moscow;Soviet Union;2;0;2  1980        (BRA)   
185   1896;(GBR);Great Britain;Athens;Greece;2;3;2  1896        (GBR)   
1121      1992;(BEL);Belgium;Barcelona;Spain;0;1;2  1992        (BEL)   
822         2008;(IRL);Ireland;Beijing;China;0;1;2  2008        (IRL)   
1308  1948;(BRA);Brazil;London;Great Britain;0;0;1  1948        (BRA)   

       Country_Name  Host_city   Host_country  Gold  Silver  Bronze  Total  
255          Brazil     Moscow   Soviet Union     2       0       2      4  
185   Great Britain     Athens         Greece     2       3       2      7  
1121        Belgium  Barcelona          Spain     0       1       2      3  
822         Ireland    Beijing          China     0       1       2      3  
1308         Brazil     London  Great Britain     0       0       1      1

In [53]:
all_time_country_medals=all_time_country_medals.drop(['all_data'],axis=1)


In [54]:
all_time_country_medals.sample(5)

Year Country_Code Country_Name       Host_city   Host_country  Gold  \
1121  1992        (BEL)      Belgium       Barcelona          Spain     0   
591   2016                     Japan  Rio de Janeiro         Brazil    12   
953   2000        (BAH)      Bahamas          Sydney      Australia     2   
377   1968        (KEN)        Kenya     Mexico City         Mexico     3   
750   2012        (AFG)  Afghanistan          London  Great Britain     0   

      Silver  Bronze  Total  
1121       1       2      3  
591        8      21     41  
953        0       1      3  
377        4       2      9  
750        0       1      1

In [55]:
all_time_country_medals.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1344 entries, 0 to 1343
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   Year          1344 non-null   object
 1   Country_Code  1344 non-null   object
 2   Country_Name  1344 non-null   object
 3   Host_city     1344 non-null   object
 4   Host_country  1344 non-null   object
 5   Gold          1344 non-null   int32 
 6   Silver        1344 non-null   int32 
 7   Bronze        1344 non-null   int32 
 8   Total         1344 non-null   int32 
dtypes: int32(4), object(5)
memory usage: 73.6+ KB


In [56]:
all_time_country_medals.duplicated().sum()

0

In [57]:
all_time_country_medals.Year.value_counts(ascending = True).head(10)

1896    11
1904    13
1908    19
1912    19
1900    21
1920    22
1924    27
1932    27
1936    32
1928    33
Name: Year, dtype: int64

In [58]:
all_time_country_medals.Country_Name.value_counts(ascending = True).tail(10)

Sweden           27
Australia        27
Canada           27
Hungary          27
Italy            27
United States    28
Switzerland      28
Denmark          28
Great Britain    29
France           29
Name: Country_Name, dtype: int64

In [59]:
all_time_country_medals.Total.value_counts(ascending = True).tail(10)

13     26
9      36
8      40
7      40
6      63
5      85
4      95
3     105
2     163
1     298
Name: Total, dtype: int64

In [60]:
all_time_country_medals.Total.value_counts(ascending = True).tail(10)

13     26
9      36
8      40
7      40
6      63
5      85
4      95
3     105
2     163
1     298
Name: Total, dtype: int64

In [61]:
# Storing the cleaed data

all_time_country_medals.to_csv('Country_Medals_all_time.csv', index=False)

<a id='conclusions'></a>
## conclusions

##### Almost, The datasets were redy and I did small cleaning operations. After cleaning the data i started to pose questions and analyze data Throght Tableau.
##### The most important findings From country medals from 1896 to 2020:

- London is the only city to host the olpmpics 3 times (1908, 1948, 2012).
- Tokyo, Los Anglos, and Athens, all hosted the Olympics 2 times.
- The most number of medals occurred in Tokyo 2020 with 1080 medals. It is the first time to be more than 1000 medals in an Olympic game.
- The USA is the most country won medals in the history of the Olympics with a total of 2630 medals( 738 Bronze, 832 Silver, and 1060 Gold) then the surprise for me the second country is the ancient soviet union with a total of 1010 medals despite it broke in 1992 which means it lost 8 Tournaments !!.

##### The most important findings From Tokyo 2020 tournament Datasets:

- The General main statistics are : Total Athletes: 11062, Female Athletes: 5432, Male Athletes: 5882, Total Coaches: 381, Total Plays: 46, Total Country Teams: 93, Total Medals: 1080.
- USA is the First country in athletes with 615 athletes, then Japan with 586 athletes.
- USA is the first country in medals with 113 medals (33 Bronze, 41 Silver, And 39 Gold), then China with 88 medals (18 Bronze, 22 Silver, And 38 Gold).
- The Athletics Play has the largest number of athletes with 2068, Swimmung with 743 athletes, then Football with 567 athletes.
- Japan has the most number of Coashes in Tokyo 2020 With 35 Coashes, Then USA with 28 Coashes, And Spain with 26 Coashes.
- Japan has the largest number of teams in Tokyo 2020 with 45 teams, then USA with 43 teams, And Italy with 34 teams. I think Japan didn't get the best result in comparison with their preparation because Japan ranked 5 in medals with 58 medals.
- Basket Ball has the most number of coashes in 2020 with 74 coashes, then Footbal with 59 coashes, And artistic swimming with 58 coashes.


#### Note: I also did the explanatory process with Tableau software to make insightful dashboards.
#### The history of Olympics dashboard:
https://public.tableau.com/app/profile/khaled.yaseen/viz/Olympics_16425314789790/TheolympicsHisyory?publish=yes

#### Tokyo 2020 dashboard:
https://public.tableau.com/app/profile/khaled.yaseen/viz/Olympics_16425314789790/Tokyo2020OlympicsGame?publish=yes
